lib/sisimai/bite/email/courier.rb

```diff
@@ -0,0 +1,265 @@
+module Sisimai::Bite::Email
+  module Courier
+    # Sisimai::Bite::Email::Courier parses a bounce email which created by Courier
+    # MTA. Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/Courier.pm
+      require 'sisimai/bite/email'
+
+      # http://www.courier-mta.org/courierdsn.html
+      # courier/module.dsn/dsn*.txt
+      Re0 = {
+        :'from'       => %r/Courier mail server at /,
+        :'subject'    => %r{(?:
+           NOTICE:[ ]mail[ ]delivery[ ]status[.]
+          |WARNING:[ ]delayed[ ]mail[.]
+          )
+        }x,
+        :'message-id' => %r/\A[<]courier[.][0-9A-F]+[.]/,
+      }
+      Re1 = {
+        :begin  => %r{(?:
+           DELAYS[ ]IN[ ]DELIVERING[ ]YOUR[ ]MESSAGE
+          |UNDELIVERABLE[ ]MAIL
+          )
+        }x,
+        :rfc822 => %r{\AContent-Type:[ ]*(?:
+           message/rfc822
+          |text/rfc822-headers
+          )\z
+        }x,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReFailure = {
+        # courier/module.esmtp/esmtpclient.c:526| hard_error(del, ctf, "No such domain.");
+        hostunknown: %r/\ANo[ ]such[ ]domain[.]\z/x,
+        # courier/module.esmtp/esmtpclient.c:531| hard_error(del, ctf,
+        # courier/module.esmtp/esmtpclient.c:532|  "This domain's DNS violates RFC 1035.");
+        systemerror: %r/\AThis[ ]domain's[ ]DNS[ ]violates[ ]RFC[ ]1035[.]\z/x,
+      }
+      ReDelayed = {
+        # courier/module.esmtp/esmtpclient.c:535| soft_error(del, ctf, "DNS lookup failed.");
+        networkerror: %r/\ADNS[ ]lookup[ ]failed[.]\z/x,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Courier MTA'; end
```

Comment:
__[rubocop]__
* Style/SingleLineMethods: Avoid single-line method definitions. [:ref](https://github.com/bbatsov/ruby-style-guide#no-single-line-methods)
* Style/Semicolon: Do not use semicolons to terminate expressions. [:ref](https://github.com/bbatsov/ruby-style-guide#no-semicolon)
* Layout/TrailingWhitespace: Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

<!-- Comment by [SideCI](https://sideci.com) -->